highway.tex

```diff
@@ -615,9 +630,207 @@ \subsection{Checkpoints}
+
+Unfortunately, it is non-trivial to deal with such attacks. One cannot simply ignore votes from a validator who is sending too many of them. For one, the validator who is originating the votes may not be the one who you are receiving them from. If these votes are contained in the justification of a block sent by a confederate, it may not be clear whether the confederate is actively assisting the denial of service attack or if they simply had a greater download capacity than you do.
+
+This problem is made worse by noting that the honest validators must essentially agree on which votes they are going to process. If validator $A$ downloads one of these spammy votes and incorporates it into one of their own justifications, other validators will not be able to process $A$'s vote until they have also processed these other messages. This is perhaps even worse if an attacker sends different sets of spam votes to each of the honest validators. If each honest validator includes a reasonable subset of these messages in their next block, it may not be apparent that the system is being overtaxed until they have a chance to look at each others' blocks and realize that they will have a huge number of new votes that need to be downloaded and processed before they can continue.
+
+In this section, we first outline two of possible spam patterns that creates very high amounts of messages, in order to show that without taking any preventive measures, the spam attacks can cause a resource depletion in validator nodes and hence possibly crash the whole protocol. Then, we show how using relatively simple techniques, it is possible to bound the amount of possible spam to the easily managable size. 
+
+\subsection{Classification of Basic Spam Patterns}
+
+In this subsection, we lay out two basic categories of spam attacks:
+
+In a vertical spam, the attacker does not produce any equivocations and merely produces a single long string of sequential messages. This could be done by simply setting their round exponent to an unrealistically small number. This is substantially the easier of type of spam to deal with, but the only type that can be produced without the attacker equivocating, and thus allowing them to be readily punished for it.
+
+In an equivocation bomb, instead of making many sequential messages, the attacker makes a large number of non-sequential messages. This is a somewhat trickier situation to deal with especially because as discussed above, the attacker can show different messages to different validators, and it will take some coordination between the other validators in order to determine which messages should and shouldn't be ignored.
```

Comment:
We could also call it a "fork bomb". And we should refer to relevant section in the Aleph paper here.

Reply:
I'm adding the second name (fork bomb), but I'm leaving the 'equivocation bomb' as the main name, as the term equivocation was used through the paper